enterprise/cmd/frontend/userlimitchecker/limitchecker.go

```diff
@@ -0,0 +1,102 @@
+package userlimitchecker
+
+import (
+	"context"
+	"time"
+
+	ps "github.com/sourcegraph/sourcegraph/enterprise/cmd/frontend/internal/dotcom/productsubscription"
+	"github.com/sourcegraph/sourcegraph/internal/database"
+	"github.com/sourcegraph/sourcegraph/internal/txemail"
+	"github.com/sourcegraph/sourcegraph/internal/types"
+	"github.com/sourcegraph/sourcegraph/lib/errors"
+)
+
+// send email to site admins if approaching user limit on active license
+func sendApproachingUserLimitAlert(ctx context.Context, db database.DB) error {
+	userCount, err := getUserCount(ctx, db)
+	if err != nil {
+		return errors.Wrap(err, "could not get user count")
+	}
+
+	userLimit, err := getLicenseUserLimit(ctx, db)
+	if err != nil {
+		return errors.Wrap(err, "could not get license user limit")
+	}
+
+	siteAdminEmails, err := getSiteAdminEmails(ctx, db)
+	if err != nil {
+		return errors.Wrap(err, "could not get site admins")
+	}
+
+	replyTo := "support@sourcegraph.com"
+	messageId := "approaching_user_limit"
+
+	if approachingOrOverUserLimit(userCount, userLimit) {
```

Comment:
```suggestion
	if approachingOrOverUserLimit(userCount, userLimit) {
		siteAdminEmails, err := getSiteAdminEmails(ctx, db)
		if err != nil {
			return errors.Wrap(err, "could not get site admins")
		}
		replyTo := "support@sourcegraph.com"
		messageId := "approaching_user_limit"
```
Nit: check for site admin emails and instantiate variables only when necessary